marketplace-and-integrations/packages/authorized-services.md

```diff
@@ -0,0 +1,776 @@
+
+There are similarities to the flow that needs to be implemented for different services.  Steps include:
+
+- Redirecting to an authentication endpoint.
+- Handling the response including an authentication code and exchanging it for an access token.
+- Securely storing the token.
+- Including the token in API requests.
+- Serializing requests and deserializing the API responses.
+- Handling cases where the token has expired and obtaining a new one via a refresh token.
+
+There are though also differences, across request and response structures and variations in the details of the flow itself.
+
+The package tries to implement a single, best practice implementation of working with OAuth. For particular providers the specific flow required can be customized via configuration or code.
+
+## Features
+
+For the solution developer, the Umbraco Authorized Services offers two primary features.
+
+Firstly there's a tree available in the _Settings_ section of the backoffice, called _Authorized Services_. The tree shows the list of services based on the details provided in configuration.
+
+![authorized-services-tree](images/authorized-services-tree.png)
+
+Each tree entry has a management screen where an administrator can authenticate with an app that has been setup with the service. The status of each service is shown on this screen. When authorized, the authentication and authorization flow has been completed and an access token stored.
+
+![not-authorized-screen](images/not-authorized-screen.png)
+
+![authorized-screen](images/authorized-screen.png)
+
+Secondly, the developer has access to an interface - `IAuthorizedServiceCaller` - that they can inject instances of and use to make authorized requests to the service's API.
+
+Using a settings screen the administrator can review the service configuration.
+
+![settings-screen](images/settings-screen.png)
+
+## Usage
+
+### App Creation
+
+Services that this package are intended to support will offer an OAuth authentication and authorization flow against an "app". The developer will need to create this "app" with the service.  By doing this information such as the "client ID" and "client secret" can be applied in configuration.
+
+### Installation
+
+The package can be installed into your Umbraco solution from [NuGet](https://www.nuget.org/packages/Umbraco.AuthorizedServices).
+
+### Configuring a Service
+
+Details of services available need to be applied to the Umbraco web application's configuration, which, if using the `appSettings.json` file, will look as follows. Other sources such as environment variables can also be used, as per standard .NET configuration.
```

Comment:
Just a slight rewrite, to avoid having "will look as follows" in the middle of a paragraph. Let me know if this doesn't work in the context!

```suggestion
Details of services available need to be applied to the Umbraco web application's configuration.

An example of doing this through the `appSettings.json` file is shown below. Other sources such as environment variables can also be used, as per standard .NET configuration.
```